### PR TITLE
cilium: add repo for cilium

### PR DESCRIPTION
**Pull request by nirmoy**
note: we dont have any caasp_registry for cilium yet. so this PR wont work now, until we have cilium in caasp_registry.

### CODE REVIEW

**Comment by nirmoy** [comment]
rebased

**Comment by nirmoy** [comment]
unrelated error:
++ kubectl --kubeconfig /srv/jenkins/workspace/salt.integration_PR-695/kubeconfig get pods --no-headers --namespace helm-392f5e5e-4a2b-442c-9723-33d134e358d0

+ POD_STATUS='helm-392f5e5e-4a2b-442c-9723-33d134e358d0-mariadb-b9846bbfn7492   0/1   Init:ImagePullBackOff   0     44s'

+ [[ -z helm-392f5e5e-4a2b-442c-9723-33d134e358d0-mariadb-b9846bbfn7492   0/1   Init:ImagePullBackOff   0     44s ]]

+ PODS_FOUND=1

+ echo 'helm-392f5e5e-4a2b-442c-9723-33d134e358d0-mariadb-b9846bbfn7492   0/1   Init:ImagePullBackOff   0     44s'

+ grep -v Running

helm-392f5e5e-4a2b-442c-9723-33d134e358d0-mariadb-b9846bbfn7492   0/1   Init:ImagePullBackOff   0     44s

+ echo 'INFO: Waiting for pods to enter running state'

INFO: Waiting for pods to enter running state

**Comment by nirmoy** [comment]
@bergmannf please review again

File: pillar/cni.sls [hunk 1]
```diff
@@ -16,6 +16,9 @@ flannel:
 # 7 - Display HTTP request headers.
 # 8 - Display HTTP request contents.
   log_level:      '2'
+# cilium configuration
+cilium:
+  image:          '{{ salt.caasp_registry.base_image_url() }}/cilium:1.2.1'
```
Comment (lcavajani) [inline pillar/cni.sls]: @inercia was suggesting to not hardcode the image version here but to add a pillar in [params.sls](https://github.com/kubic-project/salt/blob/master/pillar/params.sls)

See https://github.com/kubic-project/salt/pull/747#discussion_r258829813

Comment (inercia) [reply pillar/cni.sls]: Nice suggestion, but Salt has a terrible limitation: you cannot use pillars when defining pillars. So we must define the `image` here or in the `params.sls`, but it cannot be built as a mix.

Comment (inercia) [reply pillar/cni.sls]: I think you cannot (or should not) use `salt.caasp_registry.base_image_url()` here. You could use it in the places where you get the `cilium:image` pillar, and specify here just the `name:tag`, ie:

```YAML
cilium:
  image:          'cilium:1.2.1'
```

Comment (nirmoy) [reply pillar/cni.sls]: @lcavajani @inercia 's suggestion make sense for that PR as we don't want to hardcode container image in a jinja file. We have pillar/cni.sls so it doesn't make sense to spread cni reference to pillar/param.sls

**Comment by nirmoy** [comment]
@inercia Thanks, updated the PR, there are two more such references I will add fix for those too 
 grep base_image_url * -R |grep sls
pillar/params.sls:pod_infra_container_image: {{ salt.caasp_registry.base_image_url() }}/pause:1.0.0
pillar/cni.sls:  image:          '{{ salt.caasp_registry.base_image_url() }}/flannel:0.9.1'

**Comment by lcavajani** [comment]
LGTM

**Comment by jordimassaguerpla** [comment]
@nirmoy @inercia : Is this ready? I see some change request by based on the comments this seems addressed, does it?

**Comment by nirmoy** [comment]
@jordimassaguerpla its done from my side, I need to  create another cleaup PR to remove `{{ salt.caasp_registry.base_image_url() }}` from flannel and infra container image

**Pull request merged**